### DISCUSSION TITLE
Screen zoom factor changes during cut and paste

### DISCUSSION THREAD

**Discussion by Donatien-Alphonse** [wxMaxima-developers/wxmaxima#1895] [General]
Hello Friends,

There's a certain editing quirk that's driving me crazy. Perhaps some knowledgeable person here can suggest a solution.

Since my vision is declining, I often magnify the view in wxMaxima, by holding down the [CTRL] key and rotating the mouse wheel. This works fine and zooms in. At other times, I prefer to see more of the document all at once, so will zoom out.

Now suppose I'm zoomed in and wish to move something from one place to another by the usual cut and paste operations. I highlight the text, then right-click to access the cut operation in the context menu. As soon as I right-click, the entire display shrinks to the default (smaller) appearance. Very annoying.

I can complete the cut and paste, of course, but then have to zoom in again afterward. So, rather than the right-click context menu being a shortcut, it becomes more tedious: cut, paste, then [CTRL]-rotate to zoom back in to a decent view.

To avoid this I've been forced to access cut and paste via the Edit menu.

In other words, the shortcut requires more actions than working through the menu. This becomes very tedious when rearranging the order of various parts within a document during the final editing.

I'm using wxMaxima 22.05.0, Maxima version 5.46.0 (x86_64-pc-linux-gnu) on Linux Mint 20.3 Cinnamon.

Any suggestions?

Many thanks,

Donatien-Alphonse

**Comment by daute** [comment]
Strange. Can't reproduce that here. But - isn't it easier to use a "screen magnifier" like 'kmag' (https://apps.kde.org/kmag/) or others? They increase the size not only of the text (which you do), but also menu entries, etc.?

**Comment by Donatien-Alphonse** [comment]
Hello Daute,

Thanks for your response. I'm surprised you weren't able to reproduce this problem. I just tried out wxMaxima 22.05.0 on a different Linux Mint machine (Asus laptop) and got the exact same response: right-clicking resets the zoom factor without asking for whatever reason.

I suspect this fault is version related and something that has crept into newer versions. As an experiment, I installed the older wxMaxima 19.07.0 from the Linux Mint repository on the first machine I wrote about. It works as expected: right-clicking opens the context menu without altering the display.

Again, as an experiment, I also tried wxMaxima 18.10.1 on an old Windows 7 machine. Once more, right-clicking works as expected.

So, version 22.05.0 seems to be the culprit. If anyone else is kind enough to try this out and compare results, just do this:

1. Type in anything.
2. Now zoom in by rolling the mouse wheel while holding [CTRL].
3. Right-click anywhere.

Does it zoom out again to some default view for no reason?

Thanks for any assistance you experts can provide.

Donatien-Alphonse

**Comment by Donatien-Alphonse** [comment]
Hi again,

For some further debugging info, I just tried version 21.05.2 on an old Windows 7 machine, and everything works fine.

So far, then, it's only version  22.05.0 on Linux Mint that causes the problem.

Donatien-Alphonse

**Comment by gunterkoenigsmann** [comment]
My guess is that the problem is the following:

- Many things in wxWidgets are identified by the type of signal they send and additionally the signal ID, which on Linux is a 32-bit-number and on Windows a 16-bit one.
- wxWidgets tells the program which is the highest ID that wxWidgets ever uses internally. When wxMaxima was started it was normal to start with the program-specific IDs after the highest of those IDs.
- If an element doesn't need to be identified by an ID wxWidgets assigns it just a pseudo-random ID.

My guess is now that the pseudo-random ID for a menu-related thing happens to coincide with the "Zoom to 100%" one.

In all new releases since autumn last year wxMaxima uses wxWidgets' new methods that allow to tell the library: "give me a new ID that doesn't coincide with anything and only re-use it if the element it is connected to no more exists" => The problem should be gone in the current source.
